src/sessions.ts

```diff
@@ -581,18 +616,45 @@ export class ClientSession
       bypassPinningCheck: true
     });
 
+    const timeoutMS =
+      typeof options?.timeoutMS === 'number'
+        ? options.timeoutMS
+        : this.timeoutContext?.csotEnabled()
+        ? this.timeoutContext.timeoutMS // refresh timeoutMS for abort operation
```

Comment:
> Is allowing a user to abort their own transaction inside of a withTransaction block permitted?

Yes it is permitted, and the withTxn helper has logic to early exit if the Txn state is already complete.

> If it is - wouldn't this automatically refresh for the abort command (is that expected behavior)?

It would, good question, I would think so since the expectation in the spec is to refresh for this operation I think it would apply for either case, in or out of the convenient helper. The reason for refresh is because the reason you _could_ be aborting is because CSOT has expired, if you don't refresh you'll never even attempt the abort. It may seem unusual at first since timeoutMS can potentially double but I think if we take the perspective of abort _always_ refreshes, at least it is consistent and leads to a safer end result (actually transmitting aborts).